archive/untar.ts

```diff
@@ -47,9 +48,9 @@ export interface TarMetaWithLinkName extends TarMeta {
   linkName?: string;
 }
 
-export interface TarHeader {
-  [key: string]: Uint8Array;
-}
+export type TarHeader = {
+  [key in UstarFields]: Uint8Array;
+};
```

Comment:
If you don't want this public type to have this slight change, we could look at creating a new non-exported type and casting the header to this new type inside getMetadata: https://github.com/denoland/deno_std/blob/b64357605cf55edc2fdd278e057d7f0b9f665195/archive/untar.ts#L256-L293